src/util/RatelimitBucket.cpp

```diff
@@ -0,0 +1,43 @@
+#include "RatelimitBucket.hpp"
+
+#include <QTimer>
+
+namespace chatterino {
+
+RatelimitBucket::RatelimitBucket(int limit, int cooldown,
+                                 std::function<void(QString)> callback)
+    : limit_(limit)
+    , cooldown_(cooldown)
+    , callback_(callback)
+{
+}
+
+void RatelimitBucket::send(QString message)
+{
+    this->queue_.append(message);
+
+    if (this->pending_ < this->limit_)
+    {
+        this->execute();
+    }
+}
+
+void RatelimitBucket::execute()
```

Comment:
Maybe change `execute` to a name which hints that the function only processes a single element like `handleOne` or `tick` or maybe something else. `execute` itself is a bit generic which makes it hard to guess what it does.

Reply:
Changed the name to `handleOne` and added a small comment about it in the header file